# Bank Loan Recommendation Model

In [1]:
#import packages
import pandas as pd
import numpy as np
from pandas import Series, DataFrame
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import classification_report

### Load the data

In [2]:
#load the raw data into a Pandas dataframe
bank_df = pd.read_csv('Downloads\\bank_dataset.csv')
bank_df.head()

age  ... divorced                    
0   58  ...                            0
1   44  ...                            1
2   33  ...                            0
3   47  ...                            0
4   33  ...                            1

[5 rows x 37 columns]

### Use .info() to check the length and datatypes

In [3]:
bank_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45211 entries, 0 to 45210
Data columns (total 37 columns):
age                             45211 non-null int64
job                             45211 non-null object
marital                         45211 non-null object
education                       45211 non-null object
default                         45211 non-null object
balance                         45211 non-null int64
housing                         45211 non-null object
loan                            45211 non-null object
contact                         45211 non-null object
day                             45211 non-null int64
month                           45211 non-null object
duration                        45211 non-null int64
campaign                        45211 non-null int64
pdays                           45211 non-null int64
previous                        45211 non-null int64
poutcome                        45211 non-null object
y                               45

### Split data into train and test data set

In [7]:
X = bank_df.ix[:,(18, 19, 20, 21, 22, 23, 24, 25,26, 27, 28, 29, 30, 31, 32)].values
y = bank_df.iloc[:,17].values

C:\Users\mauli\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: FutureWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#ix-indexer-is-deprecated
  """Entry point for launching an IPython kernel.


### Logistic Regression

In [8]:
#Create logistic regression
LR = LogisticRegression()
LR.fit(X, y)
y_pred = LR.predict(X)

C:\Users\mauli\Anaconda3\lib\site-packages\sklearn\linear_model\logistic.py:432: FutureWarning: Default solver will be changed to 'lbfgs' in 0.22. Specify a solver to silence this warning.
  FutureWarning)


In [9]:
print(classification_report(y, y_pred))

              precision    recall  f1-score   support

           0       0.90      0.99      0.94     39922
           1       0.67      0.17      0.27      5289

    accuracy                           0.89     45211
   macro avg       0.79      0.58      0.61     45211
weighted avg       0.87      0.89      0.86     45211



Here, out of the entire dataset, 87% of the recommended loan offers were the offers that the users actually liked and 89% of the user's preferred loan offers were recommended.